src/mongocxx/test/spec/unified_test_format.cpp

```diff
@@ -61,22 +58,25 @@ std::vector<int> get_version(bsoncxx::document::element doc) {
     return get_version(doc.get_string().value.to_string());
 }
 
-template <typename R1, typename R2, typename Compare>
-bool is_compatible_version(R1 rng1, R2 rng2, Compare comp) {
+template <typename Range1, typename Range2, typename Compare>
+bool is_compatible_version(Range1 range1, Range2 range2, Compare comp) {
     // only compare major and minor in version of the form "<int>.<int>.<int>", i.e., [0:2)
-    return std::lexicographical_compare(
-        std::begin(rng1), std::begin(rng1) + 2, std::begin(rng2), std::begin(rng2) + 2, comp);
+    return std::lexicographical_compare(std::begin(range1),
+                                        std::begin(range1) + 2,
+                                        std::begin(range2),
+                                        std::begin(range2) + 2,
+                                        comp);
```

Comment:
The comparison logic may need to be tweaked slightly to account for different incompatible major versions.

If the runner only supports version `2.a.b`, then it should not consider a test of version `1.y.z` compatible.

https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#test-runner-support
> Test files are considered compatible with a test runner if their schemaVersion is less than or equal to a supported version in the test runner, given the same major version component.

As an aside: I like to think semver major version bumps as having nothing to do with the prior version. A major version bump does not guarantee any similarity. In an extreme case, a `4.0.0` C++ driver could completely remove all of the existing public API and replace it, and that would be valid by semver.

Reply:
Done. See `is_compatible_schema_version`